app/src/main/java/org/mozilla/fenix/home/collections/TabInCollectionViewHolder.kt

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.collections
+
+import android.view.View
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.ComposeView
+import androidx.lifecycle.LifecycleOwner
+import androidx.recyclerview.widget.RecyclerView
+import mozilla.components.feature.tab.collections.Tab
+import mozilla.components.feature.tab.collections.TabCollection
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.home.sessioncontrol.CollectionInteractor
+
+/**
+ * [RecyclerView.ViewHolder] for displaying an individual [Tab].
+ * Clients are expected to use [bindSession] to link a particular [Tab] to be displayed
+ * otherwise this will be an empty, 0 size View.
+ *
+ * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
+ * @param viewLifecycleOwner [LifecycleOwner] to which this Composable will be tied to.
+ * @param interactor [CollectionInteractor] callback for user interactions.
+ */
+class TabInCollectionViewHolder(
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: CollectionInteractor,
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
+    private var tabData = TabInfo()
+
+    @Composable
+    override fun Content() {
+        val tabInfo = remember { mutableStateOf(tabData) }
+
+        tabInfo.value.tab?.let { validTab ->
+            tabInfo.value.collection?.let { validCollection ->
+                val horizontalPadding =
+                    composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+                composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
+
+                TabInCollection(
+                    tab = validTab,
+                    isLastInCollection = tabInfo.value.isLastInCollection,
```

Comment:
Maybe we just determine the `shape` here?

Reply:
As above allowing to pass shapes leads to brittle composables that can easily be broken by clients passing a "wrong" shape.
I feel like this is a good indication that we shouldn't. Only pass indications / state on what needs to be shown and leave this Composable applying it's business rules on adapting that and in this ensuring consistency with the rest of "hardcoded" properties it uses.

Reply:
If this exact piece of UX (represented by this Composable) will be needed in other places later maybe we can just "promote" it to the `compose` package and change the naming of the `isLastInCollection` parameter but otherwise if there's any need for other shapes then other properties will probably have to be changed also. 

(Still thinking about passing a circle or some kind of an irregular shape. Shape's middle name is "Danger".)

I guess most we could do is allowing to pass a "cornerRadius" value which we can confine in between safe values. If this will be needed later.